packages/forms/src/utils/getFieldSchemaName.js

```diff
@@ -0,0 +1,13 @@
+// @flow
+const getFieldSchemaName = (name: string): string => {
+  let clearedName = name.replace(/\[\d+\]/g, '');
+
+  if (/\.?([^.]+)$/.test(clearedName)) {
+    // $FlowFixMe
+    clearedName = RegExp.$1;
```

Comment:
Not clearly behavior. Maybe it's worth adding some unit-tests for this.

Reply:
@ej9x Positive cases already tested by high-level tests